lua/orgmode/config/defaults.lua

```diff
@@ -31,6 +31,13 @@ return {
   org_custom_exports = {},
   org_indent_mode = 'indent',
   org_time_stamp_rounding_minutes = 5,
+  org_edit_src_block = {
```

Comment:
Lets call this `org-src-window-setup` and set it up to directly be a string (or a function), instead of nesting it as an `open` property.
Default value should be `top 16new`, but it can accept the function as well. Do the `:e tempname()` yourself once this is called, because that needs to happen anyway.

This also removes the need for a function to return anything.

Reply:
There might be scenarios where the user might want to handle creating the temporary file (say to have the file in a project such that the language server can read contextual type definitions, etc.), but for now this will likely work just fine. Another option would be to have the `string` option be the "simple" case, and the `function` option rely on the user to do _all_ setup of the buffer (except perhaps the wipeout handling and setting of the file content). Removing the file on disk would also have to be up to the user in this case, but handing around a callback isn't super hard to do.

Either way, I've changed it to be handled as mentioned here for now. It'd be trivial to change should the feature be needed in the future.

Reply:
Do you think `org_edit_special_window_setup` might be better?

Reply:
I guess we could have a function do everything, and string be a simple case. Function should be well documented if we take that route.

Regarding the name, I got it from here https://orgmode.org/manual/Editing-Source-Code.html#:~:text=org-src-window-setup .